saleor/payment/forms.py

```diff
@@ -0,0 +1,83 @@
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import pgettext_lazy
+
+from . import ChargeStatus, TransactionType, get_provider
+from .models import Payment
+from .utils import gateway_authorize, gateway_capture, gateway_refund
+
+
+def get_form_for_payment(payment):
+    if payment.variant == settings.DUMMY:
+        return DummyPaymentForm
+    elif payment.variant == settings.BRAINTREE:
+        return BraintreePaymentForm
+    raise ValueError('Unknown payment provider')
+
+
+class DummyPaymentForm(forms.Form):
```

Comment:
That's really odd and sounds hard to maintain/ understand because we don't have an abstract base.

Reply:
Good point, I'll create an abstract base for this